cloud/shared/bin/lib/variable_definition_loader_test.py

```diff
@@ -0,0 +1,50 @@
+import unittest
+import json
+import os
+
+from cloud.shared.bin.lib.variable_definition_loader import VariableDefinitionLoader
+"""
+ Tests for the VariableDefinitionLoader. Tests the loading of configuration
+ variables from a json file.
+
+ To run the tests from the root direcory of the deploy-infra repostory run:
+    PYTHONPATH="${PYTHONPATH}:${pwd}" 
+    python3 cloud/shared/bin/lib/variable_definition_loader_test.py
+"""
+
+class TestVariableDefinitionLoader(unittest.TestCase):
+
+    __valid_variable_defs = {
+            "BING": {
+                "required": True,
+                "secret": False,
+                "type": "string"
+            },
+            "BONG": {
+                "required": True,
+                "secret": False,
+                "type": "string"
+            },
+            "BOB": {
+                "required": False,
+                "secret": False,
+                "type": "string"
+            },
+        }
+
+    def test_load_valid_variable_definitions(self):   
+        self.__json_file_path_name = "./valid_vars_testfile.json"
```

Comment:
You can use tempfile.NamedTemporaryFile() to create these files: https://docs.python.org/3/library/tempfile.html

```
with tempfile.NamedTemporaryFile(mode="w") as f:
    self._write_json_file(self.__valid_variable_defs, f)
    varDefLoader.load_definition_file(f.name)
...
```